File: python/validParenthesis.py
def isValid(s: str) -> bool:
    stack: list[str] = []
    for val in s:
        if val == "(":
            stack.append(val)
        elif val == "[":
            stack.append(val)
        elif val == "{":
            stack.append(val)
        else:
            pop = stack.pop() if len(stack) > 0 else ""
            if pop == "":
                return False
            elif pop == "(":
                if val != ")":
                    return False
            elif pop == "{":
                if val != "}":
                    return False
            else:
                if pop != "[":
                    return False

    return len(stack) == 0


print(isValid("[({})]"))
print(isValid("[({))]"))

File: go/ValidParenthesis/ValidParenthesis.go
package main

import "fmt"

func isvalid(s string) bool {
	stack := make([]byte, 0)
	for i := 0; i < len(s); i++ {
		val := s[i]
		switch val {
		case '[':
			stack = append(stack, '[')
		case '(':
			stack = append(stack, '(')
		case '{':
			stack = append(stack, '{')
		default:
			var newVal byte
			if len(stack) > 0 {
				newVal = stack[len(stack)-1]
			} else {
				return false
			}
			switch newVal {
			case '[':
				if val != ']' {
					return false
				}
				stack = stack[:len(stack)-1]
			case '(':
				if val != ')' {
					return false
				}
				stack = stack[:len(stack)-1]
			case '{':
				if val != '}' {
					return false
				}
				stack = stack[:len(stack)-1]
			}
		}
	}
	return len(stack) == 0
}

func main() {
	fmt.Println(isvalid("[({})]"))
	fmt.Println(isvalid("[({))]"))
}
